go/cmd/dolt/commands/sqlserver/server.go

```diff
@@ -357,6 +363,30 @@ func Serve(
 	return
 }
 
+type remotesapiAuth struct {
+	ctx          *sql.Context
+	serverConfig ServerConfig
+	sqlEngine    *engine.SqlEngine
+}
+
+func newAuthenticator(ctx *sql.Context, serverConfig ServerConfig, sqlEngine *engine.SqlEngine) remotesrv.Authenticator {
+	return &remotesapiAuth{ctx, serverConfig, sqlEngine}
+}
+
+func (r *remotesapiAuth) Authenticate(ctx context.Context, creds *remotesrv.RequestCredentials) bool {
+	if creds == nil {
+		return true
+	}
+	if r.serverConfig.User() == creds.Username {
+		return r.serverConfig.Password() == creds.Password
+	}
+
+	r.ctx.Session.SetClient(sql.Client{User: creds.Username, Address: "localhost", Capabilities: 0})
```

Comment:
This seems wrong to me. The caller isn't authenticated as the calling username unless they present credentials which match with the credentials stored in MySQLDb, but we don't seem to check those here. Am I just reading it wrong?